src/Common/PODArray.h

```diff
@@ -186,11 +186,11 @@ class PODArrayBase : private boost::noncopyable, private TAllocator    /// empty
     template <typename ... TAllocatorParams>
     void reserveForNextSize(TAllocatorParams &&... allocator_params)
     {
-        if (empty())
+        if (unlikely(empty()))
         {
             // The allocated memory should be multiplication of ELEMENT_SIZE to hold the element, otherwise,
             // memory issue such as corruption could appear in edge case.
-            realloc(std::max(integerRoundUp(initial_bytes, ELEMENT_SIZE),
+            alloc(std::max(integerRoundUp(initial_bytes, ELEMENT_SIZE),
```

Comment:
AFAIU, this is not equivalent to previous code. `realloc` call `alloc` only if `c_start == null` but `empty()` means `c_start == c_end`.
So, calling `reserveForNextSize()` consequently a couple times would lead to a leak.

Reply:
Yes u r right.